consumer/consumererror/statuserrors.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"fmt"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+)
+
+// StatusError contains either an HTTP or gRPC status code
+// resulting from a failed request.
+// It allows retrieving a type of error code, converting
+// between the status codes supported by each transport if necessary.
+//
+// It should be created with NewHTTPStatus or NewGRPCStatus.
+type StatusError struct {
```

Comment:
nit: `component` has the concept of `Status` already and some of the status will have a relationship to an error.  Should we pick a different keyword here?

Reply:
I think that's a good callout. At the risk of making it too generic, I've renamed this to `NetworkError`.

Reply:
`NetworkError` works for me